nix/marlowe-symbolic-lambda.nix

```diff
@@ -0,0 +1,41 @@
+{ pkgs, haskellPackages }:
+let
+  # NOTE: Unfortunately we cannot build a static executable if the package uses template haskell.
+  # In order to get around this we make sure that the TH code is all in libraries and create
+  # a single haskell file that does nothing other than import the main function from the library.
+  # We can then statically compile this as it does not use TH.
+  staticHaskellPackages = (haskellPackages.override (old: {
+    pkgsGenerated = pkgs.callPackage ../pkgs {};
+  }));
+  ghc = staticHaskellPackages.ghcWithPackages (p: [ p.marlowe-symbolic ]);
+  main = pkgs.writeText "app.hs"
+              ''
+              module Main where
+              import qualified App
+              main = App.main
+              '';
+  z3 = pkgs.callPackage ./nix/z3.nix { staticbin = true; };
+  killallz3 = pkgs.writeScriptBin "killallz3" ''
+    kill -9 $(ps aux | grep z3 | grep -v grep | awk '{print $2}')
+  '';
+in
+  pkgs.stdenv.mkDerivation {
+    name = "marlowe-symbolic-lambda";
+    buildInputs = [ pkgs.zip ];
+    unpackPhase = "true";
+    buildPhase =
+      ''
+      mkdir -p $out/bin
+      ${ghc}/bin/ghc ${main} -static -threaded -o $out/bin/bootstrap \
+                     -optl=-static \
+                     -optl=-L${pkgs.ncurses.override { enableStatic = true; }}/lib \
+                     -optl=-L${pkgs.zlib.static}/lib \
+                     -optl=-L${pkgs.gmp6.override { withStatic = true; }}/lib \
+                     -optl=-L${pkgs.libffi.overrideAttrs (old: { dontDisableStatic = true; })}/lib
```

Comment:
Do we need all these package overrides given that we're using the `pkgsMusl` set? It doesn't work to just stick them in `buildInputs`?

Reply:
musl doesn't build things statically, it just makes it easier, so we still need to override. Or do you mean pass in a `static` flag to the nix expression and have them conditional?